File: go/1431_Kids_With_the_Greatest_Number_of_Candies.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
    var candiesMax = max(candies)
    var result []bool

    for _, candy := range candies {
        result = append(result, candy + extraCandies >= candiesMax)
    }

    return result
}

func max(candies []int) int {
    var max = 0

    for _, candy := range candies {
        if max < candy {
            max = candy
        }
    }
    
    return max
}

File: php/1400/1431_Kids_With_the_Greatest_Number_of_Candies.php
<?php

class Solution
{

    /**
     * @param Integer[] $candies
     * @param Integer $extraCandies
     * @return Boolean[]
     */
    function kidsWithCandies($candies, $extraCandies)
    {
        $candiesMax = max($candies);
        $result = [];

        foreach ($candies as $candy) {
            $result[] = $candy + $extraCandies >= $candiesMax;
        }

        return $result;
    }

}
